tree/30_generic_methods/service_check_started_at_boot.cf

```diff
@@ -47,28 +50,27 @@ bundle agent service_check_started_at_boot(service_name)
 
   classes:
 
-      "pass2"               expression => "pass1";
-      "pass1"               expression => "any";
-
       # This is to workaround a bug in Debian Jessie, that makes systemctl is-enabled
       # fail if run against a service that still uses SysV style scripts
       # See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=760616
       "init_script_present" expression => fileexists("/etc/init.d/${service_name}");
       "broken_systemctl"    expression => "debian_8.init_script_present";
 
-  methods:
-
-    pass2::
+    pass1::
+      "set_at_boot"         expression => returnszero("${command_to_check}", "useshell");
```

Comment:
In cfengine 3.6+ the results of returnszero are cached so that they are only run once. 

However this shouldn't have much impact because this is just a check command : it won't change anything and is cheap to run.